ui-particles-angular/projects/ui-particles-angular/src/lib/gio-submenu/gio-submenu.component.ts

```diff
@@ -13,11 +13,48 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Component } from '@angular/core';
+import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+
+import { GioMenuService } from '../gio-menu/gio-menu.service';
 
 @Component({
   selector: 'gio-submenu',
   templateUrl: './gio-submenu.component.html',
   styleUrls: ['./gio-submenu.component.scss'],
 })
-export class GioSubmenuComponent {}
+export class GioSubmenuComponent implements OnInit {
+  public reduced = false;
+  public overlay = false;
+  private hover = false;
+  @ViewChild('gioSubmenu', { static: false })
+  private gioSubmenu: ElementRef<HTMLDivElement> | undefined;
+
+  constructor(private readonly gioMenuService: GioMenuService) {}
+
+  public ngOnInit(): void {
+    this.gioMenuService.reduce.subscribe(reduced => {
+      this.reduced = reduced;
+    });
+    this.gioMenuService.mouseOver.subscribe(mouseOver => {
```

Comment:
wdyt to add takeUntil to close this observable on destroy this component ?

Reply:
same for all subscribe 🙈

Reply:
✅